spec/lib/jobs/dell-wsman-simple-update-firmware-spec.js

```diff
@@ -0,0 +1,278 @@
+            .then(function() {
+                var error = new Error("Failed to update firmware. Server: 000.000.000.000");
+                expect(wsmanBaseJob.prototype._done).to.be.calledWith(error);
+            });
+        });
+
+        it('should handle: response status is undefined', function() {
+            var response = {
+                body: {
+                    status: undefined
+                }
+            };
+            wsmanTool.prototype.clientRequest.resolves(response);
+            simpleUpdateFirmwareJob.targetConfig = {
+                serverAddress: '000.000.000.000'
+            };
+            return simpleUpdateFirmwareJob.clientRequest('http://localhost:46010','/api/1.0/server/firmware/updater/dup','POST', {})
+            .then(function() {
+                var error = new Error("Failed to update firmware. Server: 000.000.000.000");
+                expect(wsmanBaseJob.prototype._done).to.be.calledWith(error);
+            });
+        });
+
+        it('should handle: response status is empty', function() {
+            var response = {
+                body: {
+                    status: ''
+                }
+            };
+            wsmanTool.prototype.clientRequest.resolves(response);
+            simpleUpdateFirmwareJob.targetConfig = {
+                serverAddress: '000.000.000.000'
+            };
+            return simpleUpdateFirmwareJob.clientRequest('http://localhost:46010','/api/1.0/server/firmware/updater/dup','POST', {})
+            .then(function() {
+                var error = new Error("Failed to update firmware. Server: 000.000.000.000");
+                expect(wsmanBaseJob.prototype._done).to.be.calledWith(error);
+            });
+        });
+
+        it('should handle: error occurred while sending request in wsman tool', function() {
+            wsmanTool.prototype.clientRequest.rejects('fake error');
+            return simpleUpdateFirmwareJob.clientRequest('http://localhost:46010','/api/1.0/server/firmware/updater/dup','POST', {})
+            .then(function() {
+                var error = new Error('fake error');
+                expect(wsmanBaseJob.prototype._done).to.be.calledWith(error);
+            });
+        });
+    });
+
+    describe('DeleteDownloadImage', function() {
+        beforeEach(function() {
+            this.sandbox.stub(ChildProcess.prototype, '_parseCommandPath').returns('fake script file path');
+            fs.existsSync.resolves(true);
+        });
+
+        it('should delete download image and package.xml correcly', function() {
+            ChildProcess.prototype.run.resolves();
+            simpleUpdateFirmwareJob.DeleteDownloadImage();
+            expect(ChildProcess.prototype.run).to.have.been.called.twice;
+        });
+
+        it('should handle: error occured while processing command', function() {
+            errorLoggerSpy.reset();
+            ChildProcess.prototype.run.throws('fake error');
```

Comment:
The ChildProcess is designed to return a promise, but you mocked it as a sync function, this doesn't align with  the real world.